region/brinstar/green/Brinstar Pre-Map Room.json

```diff
@@ -229,26 +201,79 @@
     },
     {
       "link": [2, 1],
-      "name": "Speedball",
+      "name": "Speedball (Come In Shinecharging)",
+      "notable": false,
+      "entranceCondition": {
+        "comeInShinecharging": {
+          "length": 11,
+          "openEnd": 0
+        }
+      },
+      "requires": [
+        "canSpeedball"
+      ],
+      "clearsObstacles": ["A"],
+      "devNote": "There are 3 unusable tiles in this runway."
+    },
+    {
+      "link": [2, 1],
+      "name": "Speedball (In-Room)",
       "notable": false,
       "requires": [
-        "canSpeedball",
-        {"or": [
-          {"canComeInCharged": {
-            "framesRemaining": 180,
-            "fromNode": 2,
-            "unusableTiles": 3
-          }},
-          {"and": [
-            "canSlowShortCharge",
-            {"canShineCharge": {
-              "usedTiles": 14,
-              "openEnd": 1
-            }}
-          ]}
-        ]}
+        "canSlowShortCharge",
+        {"canShineCharge": {
+          "usedTiles": 14,
+          "openEnd": 1
+        }},
+        "canSpeedball"
       ],
-      "clearsObstacles": ["A"]
+      "clearsObstacles": ["A"],
+      "note": "Run on the upper platform to perform the speedball.",
+      "devNote": "FIXME: This length implies 3 unusable tiles, yet the docs say 5 for a slow speedball in general?"
```

Comment:
Other slow speedballs are through door so there's a bit of weirdness.
But moreso, this is a harder shortcharge at 14 so a smaller short hop is expected.  `canSlowShortCharge` is targetted more at 16 tiles.  Really the tech doesn't need to be here because you jump as soon as you turn blue so it would be covered by doing a 14 shortcharge.

Reply:
Makes sense. Sounds like at some point there could be an opportunity to refine the requirements for other speedballs, to reflect that with a good enough shortcharge it can be doable with fewer than 5 tiles. The fact that going through a door transition can make a shortcharge harder is another thing we aren't modeling in general but maybe we could; not sure exactly what that would look like. Also shortcharges requiring heat damage -- maybe it could be useful to include a leniency multiplier (or require heatProof) to represent the need for many tries, for shortcharges close the minimum length for the selected difficulty, similar to what we are doing for other tricks like gate glitches and X-ray climb setups. On the other hand, there are some cases where shortcharges are not retriable, e.g. shinecharging out the top left door of Crocomire Escape with the gate open, so this could get tricky.

Reply:
`canShortHopMockball` is a tech that is hard to define so it is assumed that you can do shorter hops as difficulty rises.

Missing a shortcharge is related to how close it is to your selected runway length.  So it would take many tries at every level if you are going for the shortest shortcharge in logic.  And adding a few tiles makes it much easier to get every time.  

One shot tricks are expected at extreme, VeryHard-Expert would opt-in to them with notables.  Hard we're trying to avoid them for race settings.  But its still very frustrating below expert.  And any skill level can blow the shinespark even if they hit the shortcharge.  Looping around to try again adding some more costs.  But its mostly on the randomizer to keep the situation from feeling too bad.